app/src/main/java/com/cronus/zdone/home/HomeScreen.kt

```diff
@@ -115,7 +115,7 @@ class HomeScreen @Inject constructor(
                                 activity?.let {
                                     Toast.makeText(it, "Failed to mark ${task.name} as completed, reloading tasks...", Toast.LENGTH_SHORT).show()
                                 }
-                                requestTaskData() // update task to re-show task marked as completed
+                                refreshTaskData() // update task to re-show task marked as completed
```

Comment:
this will shoot off a new api request that will cause new data to be emitted on the data stream observable in `TasksRepositoryImpl`. The HomeScreen maintains a subscription to these updates (see `requestTaskData`) and will pass the updated data to the view.